### PR TITLE
wrap properties and some fsa algorithms to Python

### PR DESCRIPTION
**Pull request by qindazhu**

### CODE REVIEW

File: k2/csrc/aux_labels.h [hunk 1]
```diff
@@ -68,7 +68,7 @@ class AuxLabels1Mapper {
       : labels_in_(labels_in), arc_map_(arc_map) {}
 
   /*
-    Do enough work that know now much memory will be needed, and output
+    Do enough work to know now much memory will be needed, and output
```
Comment (csukuangfj) [inline k2/csrc/aux_labels.h:71]: to know **h**ow much

File: k2/python/k2/fsa_algo.py [hunk 2]
```diff
@@ -26,5 +29,51 @@ def get_output(self, fsa_out: Fsa, arc_map: IntArray1 = None):
...
+            fsa_out.get_base(),
+            arc_map.get_base() if arc_map is not None else None)
+
+
+class Intersection(_Intersection):
+
+    def __init__(self, fsa_a: Fsa, fsa_b: Fsa):
+        super().__init__(fsa_a.get_base(), fsa_b.get_base())
+
+    def get_sizes(self, array_size: IntArray2):
```
Comment (csukuangfj) [inline k2/python/k2/fsa_algo.py:69]: the type hint for the return value is missing.

File: k2/python/csrc/properties.cc [hunk 3]
```diff
@@ -0,0 +1,34 @@
+// k2/python/csrc/property.cc
```
Comment (csukuangfj) [inline k2/python/csrc/properties.cc:1]: properties.cc

File: k2/python/csrc/properties.cc [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#include "k2/csrc/array.h"
+#include "k2/csrc/fsa.h"
+#include "k2/csrc/properties.h"
+#include "k2/python/csrc/array.h"
+
+// We would never pass `order` parameter to k2::IsAcyclic in Python code.
+// We can make it accept `None` with `std::optional` in pybind11, but
+// that will require C++17, so we here choose to write a version without
+// `order`.
+bool IsAcyclic(const k2::Fsa &fsa) {
```
Comment (csukuangfj) [inline k2/python/csrc/properties.cc:20]: `static bool IsAcyclic`

**Comment by qindazhu** [comment]
Updated, thanks!

**Comment by danpovey** [comment]
LGTM!

**Comment by qindazhu** [comment]
Merging..

**Pull request merged**